### DISCUSSION TITLE
Chromium browser issue with chromedriver

### DISCUSSION THREAD

**Discussion by MortazaG** [ultrafunkamsterdam/undetected-chromedriver#1095] [Q&A]
Hi!
I uninstalled Chrome today and installed an older version of Chromium browser instead, version 109.
I've downloaded the proper chromedriver and at first I got this error:

> TypeError: expected str, bytes or os.PathLike object, not NoneType

Which was fixed by adding:
`options.binary_location = 'C:/Users/username/AppData/Local/Chromium/Application/chrome.exe'`

Although, using regular Selenium there was no need for setting this path. 

After setting the path to the binary file, I now get this error message:

> selenium.common.exceptions.WebDriverException: Message: unknown error: cannot connect to chrome at 127.0.0.1:56118
from session not created: 
This version of ChromeDriver only supports Chrome version 110
Current browser version is 109.0.5414.0

I have double and triple checked that I have downloaded the right version of the chromedriver.

I have set its path through:
`path = 'D:/path/to/chromedriver.exe'`
`driver = uc.Chrome(options=options, executable_path=path)`

But I am still getting the above error message.

Is there a specific way of setting up undetected_chromedriver to work properly with the Chromium browser?

**Comment by Anticope12** [comment]
Found a solution for the problem.
forget about all the extra arguments and updates/ etc.

If you are using the DIR folder that currently has a browser profile open. it will not work. you have to close everything first and then run it. That was the workaround for it for me.

**Reply by MortazaG** [reply]
Thanks for the reply! I tried what you said, but i still get the same error:

> selenium.common.exceptions.WebDriverException: Message: unknown error: cannot connect to chrome at 127.0.0.1:49812 from session not created: 
This version of ChromeDriver only supports Chrome version 110
Current browser version is 109.0.5414.0

Do you know why setting the executable_path to the chromedriver isn't working? Or what kind of special tweaking is necessary for chromium browsers as mentioned in the readme file?
> Works also on Brave Browser and many other Chromium based browsers, some tweaking

**Comment by rafaelfliu** [comment]
not sure if @MortazaG has already found the solution, but you can simply do this:

```
   import undetected_chromedriver as uc
   driver = uc.Chrome( version_main = 110 )
```
